File: platforms/hacker-rank/E--25--flatland-space-stations/solutions/solution.py
#!/bin/python3

import math
import os
import random
import re
import sys


# Complete the flatlandSpaceStations function below.
def flatlandSpaceStations(n, c):
    c.sort()

    # distance before first station
    maxd = c[0]

    # distance between stations
    for i in range(1, len(c)):
        maxd = max(maxd, (c[i] - c[i - 1]) // 2)

    # distance after last station
    maxd = max(maxd, (n - 1) - c[-1])

    return maxd


if __name__ == "__main__":
    fptr = open(os.environ["OUTPUT_PATH"], "w")
    nm = input().split()
    n = int(nm[0])
    m = int(nm[1])
    c = list(map(int, input().rstrip().split()))
    result = flatlandSpaceStations(n, c)
    fptr.write(str(result) + "\n")
    fptr.close()

File: platforms/hacker-rank/E--25--flatland-space-stations/solutions/solution.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"slices"
	"strconv"
	"strings"
)

// Complete the flatlandSpaceStations function below.
func flatlandSpaceStations(n int32, c []int32) int32 {
	slices.Sort(c)
	maxd := c[0]
	m := len(c)
	for i := 1; i < m; i++ {
		maxd = max(maxd, (c[i]-c[i-1])/2)
	}

	maxd = max(maxd, (n-1)-(c[m-1]))

	return maxd
}

func main() {
	reader := bufio.NewReaderSize(os.Stdin, 1024*1024)

	stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
	checkError(err)

	defer stdout.Close()

	writer := bufio.NewWriterSize(stdout, 1024*1024)

	nm := strings.Split(readLine(reader), " ")

	nTemp, err := strconv.ParseInt(nm[0], 10, 64)
	checkError(err)
	n := int32(nTemp)

	mTemp, err := strconv.ParseInt(nm[1], 10, 64)
	checkError(err)
	m := int32(mTemp)

	cTemp := strings.Split(readLine(reader), " ")

	var c []int32

	for i := 0; i < int(m); i++ {
		cItemTemp, err := strconv.ParseInt(cTemp[i], 10, 64)
		checkError(err)
		cItem := int32(cItemTemp)
		c = append(c, cItem)
	}

	result := flatlandSpaceStations(n, c)

	fmt.Fprintf(writer, "%d\n", result)

	writer.Flush()
}

func readLine(reader *bufio.Reader) string {
	str, _, err := reader.ReadLine()
	if err == io.EOF {
		return ""
	}

	return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}
